Favor/Favor/Sources/Scenes/Dashboard/Views/FavorTabBar.swift

```diff
@@ -0,0 +1,233 @@
+    )
+    // 6. 나머지 선분과 테두리의 CornerRadius를 적용한 모서리를 그립니다.
+    bezPath.addLine(to: CGPoint(x: self.frame.maxX - borderCornerRadius, y: 0))
+    bezPath.addCurve(
+      to: CGPoint(x: topRightPoint.x, y: borderCornerRadius),
+      controlPoint1: CGPoint(x: topRightPoint.x - borderCornerRadius + 6.63, y: 0),
+      controlPoint2: CGPoint(x: topRightPoint.x, y: 5.37)
+    )
+    bezPath.addLine(to: bottomRightPoint)
+    bezPath.addLine(to: bottomLeftPoint)
+    bezPath.addLine(to: CGPoint(x: 0, y: borderCornerRadius))
+    bezPath.addCurve(
+      to: CGPoint(x: borderCornerRadius, y: 0),
+      controlPoint1: CGPoint(x: 0, y: 5.37),
+      controlPoint2: CGPoint(x: 5.37, y: 0)
+    )
+    bezPath.close()
+    
+    return bezPath.cgPath
+  }
+  
+  /// 버튼을 만들어냅니다.
+  private func makeButton(_ icon: UIImage?) -> UIButton {
+    var config = UIButton.Configuration.plain()
+    config.baseBackgroundColor = .clear
+    config.image = icon?
+      .resize(newWidth: 24.0)
+    let button = UIButton(configuration: config)
+    button.configurationUpdateHandler = {
+      let image: UIImage? = icon?
+        .resize(newWidth: 24.0)
+      switch $0.state {
+      case .selected:
+        $0.configuration?.image = image?.withTintColor(.favorColor(.main))
+      default:
+        $0.configuration?.image = image?.withTintColor(.favorColor(.nav))
+      }
+    }
+    return button
+  }
+  
+  /// 인덱스가 바뀔 때 마다 UI를 업데이트합니다.
+  private func updateUI() {
+    self.homeButton.isSelected = false
+    self.friendButton.isSelected = false
+    if self.selectedIndex == 0 {
+      self.homeButton.isSelected = true
+    } else {
+      self.friendButton.isSelected = true
+    }
+  }
+  
+  /// 터치 영역을 조절하는 메서드입니다.
+  override func hitTest(_ point: CGPoint, with event: UIEvent?) -> UIView? {
+    if self.isHidden {
+      return super.hitTest(point, with: event)
+    }
+    let from = point
+    let to = self.middleButton.center
+    
+    // 피타고라스 법칙을 이용하여 중앙 버튼의 터치를 감지합니다.
+    return sqrt((from.x - to.x) * (from.x - to.x) + (from.y - to.y) * (from.y - to.y))
+    <= 20 ? self.middleButton :super.hitTest(point, with: event)
+  }
+  
+  /// 중앙 버튼이 터치될 때 불려지는 메서드입니다.
+  @objc private func didTapMiddleButton() {
+    self.middleButtonObserver?()
+  }
+}
+
+// MARK: - Setup
+
+extension FavorTabBar: BaseView {
+  func setupStyles() {
+    self.backgroundColor = .favorColor(.white)
+    self.barTintColor = .favorColor(.white)
+  }
+  
+  func setupLayouts() {
+    [
+      self.homeButton,
+      self.friendButton
+    ].forEach {
+      self.insertSubview($0, at: 0)
+    }
+  }
+  
+  func setupConstraints() {
+    self.homeButton.snp.makeConstraints { make in
+      make.top.equalToSuperview().inset(12.0)
+      make.leading.equalToSuperview().inset(70.0)
+    }
+    
+    self.friendButton.snp.makeConstraints { make in
+      make.top.equalToSuperview().inset(12.0)
+      make.trailing.equalToSuperview().inset(70.0)
+    }
```

Comment:
```suggestion
    self.homeButton.snp.makeConstraints { make in
      make.top.equalToSuperview().inset(12.0)
      make.centerX.equalToSuperview().multipliedBy(0.5)
    }
    
    self.friendButton.snp.makeConstraints { make in
      make.top.equalToSuperview().inset(12.0)
      make.centerX.equalToSuperview().multipliedBy(1.5)
    }
```

좌측 우측을 기준으로 레이아웃을 잡으면 화면이 작아지거나 커질 때 한 쪽으로 쏠릴 수 있을 것 같아
`centerX`를 기준으로 정렬해주는게 좋을듯..!